kratos/tests/test_mortar_mapper.py

```diff
@@ -0,0 +1,187 @@
+            "max_number_iterations"            : 10,
+            "integration_order"                : 2
+        }
+        """)
+
+        if (pure_implicit == True):
+            #linear_solver = ExternalSolversApplication.SuperLUSolver()
+            linear_solver = KratosMultiphysics.SkylineLUFactorizationSolver()
+
+            if (num_nodes == 3):
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters, linear_solver)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D3NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters, linear_solver)
+            else:
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D4NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters, linear_solver)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D4NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters, linear_solver)
+        else:
+            if (num_nodes == 3):
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D3NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters)
+            else:
+                self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D4NDoubleHistorical(self.main_model_part, KratosMultiphysics.TEMPERATURE, map_parameters)
+                self.mortar_mapping_vector = KratosMultiphysics.SimpleMortarMapperProcess3D4NVectorHistorical(self.main_model_part, KratosMultiphysics.DISPLACEMENT, map_parameters)
+
+    def _mapper_tests(self, input_filename, num_nodes, pure_implicit = False):
+
+        self.__base_test_mapping(input_filename, num_nodes, pure_implicit)
+
+        self.mortar_mapping_double.Execute()
+        self.mortar_mapping_vector.Execute()
+
+        ### DEBUG
+        #self.__post_process()
+
+        import from_json_check_result_process
+
+        check_parameters = KratosMultiphysics.Parameters("""
+        {
+            "check_variables"      : ["TEMPERATURE","DISPLACEMENT"],
+            "input_file_name"      : "",
+            "model_part_name"      : "Structure",
+            "sub_model_part_name"  : "Parts_Parts_Auto1"
+        }
+        """)
+
+        check_parameters["input_file_name"].SetString(input_filename+".json")
+
+        check = from_json_check_result_process.FromJsonCheckResultProcess(self.StructureModel, check_parameters)
+        check.ExecuteInitialize()
+        check.ExecuteBeforeSolutionLoop()
+        check.ExecuteFinalizeSolutionStep()
+
+        #import json_output_process
```

Comment:
if this is for debugging or setting up the test can you make a comment abt it?
If it is not needed I would remove it

Reply:
Ok